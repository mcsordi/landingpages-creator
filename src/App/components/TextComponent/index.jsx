import P from 'prop-types';
function TextComponent({ children, darkMode = false }) {
  return (
    <p
      className={`${darkMode ? `text-white` : `text-primary`} font-sans text-small`}
    >
      {children}
    </p>
  );
}
TextComponent.propTypes = {
  children: P.string,
  darkMode: P.bool,
};
export default TextComponent;
