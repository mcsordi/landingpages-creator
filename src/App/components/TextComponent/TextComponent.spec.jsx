import { render, screen } from '@testing-library/react';
import TextComponent from '.';

describe('<TextComponent/>', () => {
  it('should render TextComponent with "text" inside', () => {
    render(<TextComponent>text</TextComponent>);
    const textComp = screen.getByText('text');
    expect(textComp).toBeInTheDocument();
  });
  it('should render TextComponent with font-sans text-small classes', () => {
    render(<TextComponent>text</TextComponent>);
    const textComp = screen.getByText('text');
    expect(textComp).toHaveClass('text-small');
    expect(textComp).toHaveClass('font-sans');
    expect(textComp).toHaveClass('text-primary');
  });
  it('should render TextComponent with white text', () => {
    render(<TextComponent darkMode>text</TextComponent>);
    const textComp = screen.getByText('text');
    expect(textComp).toHaveClass('text-white');
  });
  it('should match a snapshot', () => {
    const { container } = render(<TextComponent>text</TextComponent>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
