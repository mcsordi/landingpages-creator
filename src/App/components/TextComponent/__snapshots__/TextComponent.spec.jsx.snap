// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent/> should match a snapshot 1`] = `
<p
  class="text-primary font-sans text-small"
>
  text
</p>
`;
